client/ayon_silhouette/plugins/load/load_source.py

```diff
@@ -30,28 +44,43 @@ def load(self, context, name=None, namespace=None, options=None):
 
         filepath = self.filepath_from_context(context)
 
-        # Add Source item to the project
-        source = fx.Source(filepath)
-
-        # Provide a nice label indicating the product
-        source.label = self._get_label(context)
-        project.addItem(source)
-
-        # property.hidden = True  # hide the attribute
-        lib.imprint(source, data={
-            "name": str(name),
-            "namespace": str(namespace),
-            "loader": str(self.__class__.__name__),
-            "representation": context["representation"]["id"],
-        })
-
-        # container = pipeline.containerise(
-        #     name=str(name),
-        #     namespace=str(namespace),
-        #     nodes=nodes,
-        #     context=context,
-        #     loader=str(self.__class__.__name__),
-        # )
+        # A source file may contain multiple parts, such as a left view
+        # and a right view in a single EXR.
+        options = options or {}
+        load_all_parts = options.get("load_all_parts", True)
+
+        has_multiple_parts = os.path.splitext(filepath)[-1].lower() in {
+            ".exr", ".sxr"
+        } and fx.Source(filepath, part=1).video
+
+        part = 0
+        while True:
+            source = fx.Source(filepath, part=part)
+
+            # Check whether the chosen part has any video or audio, if not
+            # we assume the part does not exist and we stop loading.
+            if not source.video and not source.audio:
+                if part == 0:
+                    self.log.warning("Loaded source has no video or audio.")
+                break
+
+            # Provide a nice label indicating the product
+            source.label = self._get_label(context)
```

Comment:
Preferably we can get the `[rgba_left]` and `[rgba_right]` suffix appended to the label based on the loaded part. See https://forum.borisfx.com/t/python-load-both-channels-of-stereo-exr-sxr/20215/3?u=bigroy

But it's still unclear how to get that to that name.

@jakubjezek001 any potential idea?